node/public.go

```diff
@@ -0,0 +1,66 @@
+// Package node provides a high level wrapper around tendermint services.
+package node
+
+import (
+	"fmt"
+
+	"github.com/tendermint/tendermint/config"
+	"github.com/tendermint/tendermint/libs/log"
+	"github.com/tendermint/tendermint/libs/service"
+	"github.com/tendermint/tendermint/p2p"
+	"github.com/tendermint/tendermint/privval"
+	"github.com/tendermint/tendermint/proxy"
+	"github.com/tendermint/tendermint/types"
+)
+
+// NewDefault constructs a tendermint node service for use in go process that
+// host their own process-local tendermint node. This is roughly
+// equivalent to running tendermint in it's own process communicating
+// to an external ABCI application.
+func NewDefault(conf *config.Config, logger log.Logger) (service.Service, error) {
+	return newDefaultNode(conf, logger)
+}
+
+// New constructs a tendermint node for use in an integrated
+// context. The ClientCreator makes it possible to construct.
+// The final option is a pointer to a Genesis document: if the value
+// is nil, the genesis document is read from the file specified in the
+// config, and otherwise the node uses value of the final argument.
+func New(conf *config.Config,
```

Comment:
How does an app insert a custom DB?

Reply:
via the db in the configuration instance. 

We're going to remove custom dbs entirely as part of our database project, and I expect that the interface of the node package will have to change accordingly, but I'd prefer to do that at the same time.